lib/Models/SdmxJsonCatalogItem.js

```diff
@@ -0,0 +1,931 @@
+'use strict';
+
+/*global require*/
+var URI = require('urijs');
+
+var clone = require('terriajs-cesium/Source/Core/clone');
+var defaultValue = require('terriajs-cesium/Source/Core/defaultValue');
+var defined = require('terriajs-cesium/Source/Core/defined');
+var defineProperties = require('terriajs-cesium/Source/Core/defineProperties');
+// var DeveloperError = require('terriajs-cesium/Source/Core/DeveloperError');
+var freezeObject = require('terriajs-cesium/Source/Core/freezeObject');
+// var deprecationWarning = require('terriajs-cesium/Source/Core/deprecationWarning');
+var knockout = require('terriajs-cesium/Source/ThirdParty/knockout');
+var loadJson = require('terriajs-cesium/Source/Core/loadJson');
+var when = require('terriajs-cesium/Source/ThirdParty/when');
+
+var arrayProduct = require('../Core/arrayProduct');
+// var arraysAreEqual = require('../Core/arraysAreEqual');
+var CsvCatalogItem = require('./CsvCatalogItem');
+var DisplayVariablesConcept = require('../Map/DisplayVariablesConcept');
+var inherit = require('../Core/inherit');
+var TerriaError = require('../Core/TerriaError');
+var overrideProperty = require('../Core/overrideProperty');
+var proxyCatalogItemUrl = require('./proxyCatalogItemUrl');
+var RegionMapping = require('../Models/RegionMapping');
+var TableColumn = require('../Map/TableColumn');
+var TableStructure = require('../Map/TableStructure');
+var VariableConcept = require('../Map/VariableConcept');
+var VarType = require('../Map/VarType');
+
+/*
+    The SDMX-JSON format.
```

Comment:
Should some of this text go in the jsdoc so that it is included in our generated reference documentation?  Some of it is implementation notes and need not go in the jsdoc, but parts of it seem useful to folks trying to use this class.